framework/src/modules/chain/blocks/chain.js

```diff
@@ -417,22 +417,28 @@ const popLastBlock = async (
 	slots,
 	oldLastBlock,
 	exceptions
-) => {
-	let secondLastBlock;
-	await storage.entities.Block.begin('Chain:deleteBlock', async tx => {
-		secondLastBlock = await loadSecondLastBlock(
-			storage,
-			interfaceAdapters,
-			genesisBlock,
-			oldLastBlock.previousBlock,
+) =>
+	storage.entities.Block.begin('Chain:deleteBlock', async tx => {
+		const [storageResult] = await storage.entities.Block.get(
+			{ id: oldLastBlock.previousBlock },
+			{ extended: true },
 			tx
 		);
+
+		if (!storageResult) {
+			throw new Error('PreviousBlock is null');
+		}
+
+		const secondLastBlock = storageRead(storageResult);
+		secondLastBlock.transactions = interfaceAdapters.transactions.fromBlock(
+			secondLastBlock
+		);
+
 		await undoConfirmedStep(storage, slots, oldLastBlock, exceptions, tx);
 		await backwardTickStep(roundsModule, oldLastBlock, secondLastBlock, tx);
 		await deleteBlock(storage, oldLastBlock.id, tx);
+		return secondLastBlock;
```

Comment:
does `begin` returns the result from inner function?

Reply:
Yes. I've just double checked and it does return the `secondLastBlock` in this case